diff --git b/vendor/janestreet-bignum/bigint/src/zarith.ml a/vendor/janestreet-bignum/bigint/src/zarith.ml
index 2717480..6220313 100644
--- b/vendor/janestreet-bignum/bigint/src/zarith.ml
+++ a/vendor/janestreet-bignum/bigint/src/zarith.ml
@@ -1,4 +1,3 @@
 module Big_int_Z      = Big_int_Z
 module Q              = Q
 module Z              = Z
-module Zarith_version = Zarith_version
diff --git b/vendor/janestreet-bignum/src/zarith.ml a/vendor/janestreet-bignum/src/zarith.ml
index 2717480..6220313 100644
--- b/vendor/janestreet-bignum/src/zarith.ml
+++ a/vendor/janestreet-bignum/src/zarith.ml
@@ -1,4 +1,3 @@
 module Big_int_Z      = Big_int_Z
 module Q              = Q
 module Z              = Z
-module Zarith_version = Zarith_version
