diff --git b/vendor/janestreet-incr_dom/src/js_misc.ml a/vendor/janestreet-incr_dom/src/js_misc.ml
index 65f7b44..3e09a8c 100644
--- b/vendor/janestreet-incr_dom/src/js_misc.ml
+++ a/vendor/janestreet-incr_dom/src/js_misc.ml
@@ -27,18 +27,10 @@ type rows_or_columns =
   | Columns
 [@@deriving sexp, bin_io, variants, compare]
 
-let innerHeight () =
-  Js.Optdef.case
-    Dom_html.window##.innerHeight
-    (fun () -> Dom_html.document##.documentElement##.clientHeight)
-    Fn.id
+let innerHeight () = Dom_html.window##.innerHeight
 ;;
 
-let innerWidth () =
-  Js.Optdef.case
-    Dom_html.window##.innerWidth
-    (fun () -> Dom_html.document##.documentElement##.clientWidth)
-    Fn.id
+let innerWidth () = Dom_html.window##.innerWidth
 ;;
 
 let element_is_in_viewport (elt : Dom_html.element Js.t) =
