diff --git b/vendor/janestreet-bonsai/extra/dune a/vendor/janestreet-bonsai/extra/dune
index 54250bd..63a29e7 100644
--- b/vendor/janestreet-bonsai/extra/dune
+++ a/vendor/janestreet-bonsai/extra/dune
@@ -1,3 +1,4 @@
-(library (name bonsai_extra)
+(library 
+ (name bonsai_extra)
  (libraries bonsai virtual_dom.ui_effect incr_dom.ui_incr)
- (public_name bonsai.extra) (preprocess (pps ppx_jane ppx_pattern_bind)))
\ No newline at end of file
+ (public_name bonsai.extra) (preprocess (pps ppx_jane ppx_pattern_bind)))
diff --git b/vendor/janestreet-bonsai/web_ui/not_connected_warning_box/src/dune a/vendor/janestreet-bonsai/web_ui/not_connected_warning_box/src/dune
index 28bdcad..1407372 100644
--- b/vendor/janestreet-bonsai/web_ui/not_connected_warning_box/src/dune
+++ a/vendor/janestreet-bonsai/web_ui/not_connected_warning_box/src/dune
@@ -1,3 +1,5 @@
-(library (name bonsai_web_ui_not_connected_warning_box)
+(library 
+ (name bonsai_web_ui_not_connected_warning_box)
+ (public_name bonsai.web_ui_not_connected_warning_box)
  (libraries core bonsai virtual_dom)
- (preprocess (pps js_of_ocaml-ppx ppx_jane)))
\ No newline at end of file
+ (preprocess (pps js_of_ocaml-ppx ppx_jane)))
diff --git b/vendor/janestreet-bonsai/web_ui/notifications/src/dune a/vendor/janestreet-bonsai/web_ui/notifications/src/dune
index 05a8841..91fb67c 100644
--- b/vendor/janestreet-bonsai/web_ui/notifications/src/dune
+++ a/vendor/janestreet-bonsai/web_ui/notifications/src/dune
@@ -1,3 +1,5 @@
-(library (name bonsai_web_ui_notifications)
+(library 
+  (name bonsai_web_ui_notifications)
+  (public_name bonsai.web_ui_notifications)
  (libraries bonsai bonsai_extra bonsai_web core)
- (preprocess (pps ppx_jane ppx_css)))
\ No newline at end of file
+ (preprocess (pps ppx_jane ppx_css)))
diff --git b/vendor/janestreet-bonsai/web_ui/partial_render_table/src/dune a/vendor/janestreet-bonsai/web_ui/partial_render_table/src/dune
index d77fd1d..1a93531 100644
--- b/vendor/janestreet-bonsai/web_ui/partial_render_table/src/dune
+++ a/vendor/janestreet-bonsai/web_ui/partial_render_table/src/dune
@@ -1,4 +1,5 @@
-(library (name bonsai_web_ui_partial_render_table)
- (libraries bonsai_web core incr_map_collate
-  bonsai_web_ui_element_size_hooks)
- (preprocess (pps js_of_ocaml-ppx ppx_jane ppx_pattern_bind)))
\ No newline at end of file
+(library 
+  (name bonsai_web_ui_partial_render_table)
+  (public_name bonsai.web_ui_partial_render_table)
+  (libraries bonsai_web core incr_map.collate bonsai_web_ui_element_size_hooks)
+  (preprocess (pps js_of_ocaml-ppx ppx_jane ppx_pattern_bind)))
diff --git b/vendor/janestreet-bonsai/web_ui/search_bar/src/dune a/vendor/janestreet-bonsai/web_ui/search_bar/src/dune
index 7a3b313..46433ac 100644
--- b/vendor/janestreet-bonsai/web_ui/search_bar/src/dune
+++ a/vendor/janestreet-bonsai/web_ui/search_bar/src/dune
@@ -1,2 +1,5 @@
-(library (name bonsai_web_ui_search_bar) (libraries bonsai_web core)
- (preprocess (pps ppx_jane js_of_ocaml-ppx ppx_pattern_bind)))
\ No newline at end of file
+(library 
+  (name bonsai_web_ui_search_bar) 
+  (public_name bonsai.web_ui_search_bar) 
+  (libraries bonsai_web core)
+  (preprocess (pps ppx_jane js_of_ocaml-ppx ppx_pattern_bind)))
diff --git b/vendor/janestreet-bonsai/web_ui/tabs/src/dune a/vendor/janestreet-bonsai/web_ui/tabs/src/dune
index d875d9d..31901f1 100644
--- b/vendor/janestreet-bonsai/web_ui/tabs/src/dune
+++ a/vendor/janestreet-bonsai/web_ui/tabs/src/dune
@@ -1,2 +1,5 @@
-(library (name bonsai_web_ui_tabs) (libraries bonsai_web core)
- (preprocess (pps ppx_jane ppx_pattern_bind)))
\ No newline at end of file
+(library 
+  (name bonsai_web_ui_tabs) 
+  (public_name bonsai.web_ui_tabs) 
+  (libraries bonsai_web core)
+  (preprocess (pps ppx_jane ppx_pattern_bind)))
diff --git b/vendor/janestreet-bonsai/web_ui/url_var/dune a/vendor/janestreet-bonsai/web_ui/url_var/dune
index d813534..9be9694 100644
--- b/vendor/janestreet-bonsai/web_ui/url_var/dune
+++ a/vendor/janestreet-bonsai/web_ui/url_var/dune
@@ -1,3 +1,5 @@
-(library (name bonsai_web_ui_url_var)
- (libraries bonsai bonsai_web core html5_history)
- (preprocess (pps ppx_jane js_of_ocaml-ppx)))
\ No newline at end of file
+(library 
+  (name bonsai_web_ui_url_var)
+  (public_name bonsai.web_ui_url_var)
+  (libraries bonsai bonsai_web core html5_history)
+  (preprocess (pps ppx_jane js_of_ocaml-ppx)))
