diff --git b/vendor/dbuenzli-logs/dune-project a/vendor/dbuenzli-logs/dune-project
new file mode 100644
index 0000000..42c0c16
--- /dev/null
+++ a/vendor/dbuenzli-logs/dune-project
@@ -0,0 +1 @@
+(lang dune 1.10)
diff --git b/vendor/dbuenzli-logs/src/dune a/vendor/dbuenzli-logs/src/dune
new file mode 100644
index 0000000..2851453
--- /dev/null
+++ a/vendor/dbuenzli-logs/src/dune
@@ -0,0 +1,10 @@
+(library
+  (name logs)
+  (public_name logs)
+  (modules logs))
+
+(library
+  (name logs_fmt)
+  (public_name logs.fmt)
+  (libraries logs fmt)
+  (modules logs_fmt))
diff --git b/vendor/dbuenzli-logs/src/logs.ml a/vendor/dbuenzli-logs/src/logs.ml
index bcaed2a..179a380 100644
--- b/vendor/dbuenzli-logs/src/logs.ml
+++ a/vendor/dbuenzli-logs/src/logs.ml
@@ -203,7 +203,7 @@ let pp_header ppf (l, h) = match h with
 let pp_exec_header =
   let x = match Array.length Sys.argv with
   | 0 -> Filename.basename Sys.executable_name
-  | n -> Filename.basename Sys.argv.(0)
+  | _n -> Filename.basename Sys.argv.(0)
   in
   let pf = Format.fprintf in
   let pp_header ppf (l, h) =
@@ -219,9 +219,9 @@ let format_reporter
     ?(app = Format.std_formatter)
     ?(dst = Format.err_formatter) ()
   =
-  let report src level ~over k msgf =
+  let report _src level ~over k msgf =
     let k _ = over (); k () in
-    msgf @@ fun ?header ?tags fmt ->
+    msgf @@ fun ?header ?tags:_ fmt ->
     let ppf = if level = App then app else dst in
     Format.kfprintf k ppf ("%a@[" ^^ fmt ^^ "@]@.") pp_header (level, header)
   in
diff --git b/vendor/dbuenzli-logs/src/logs_fmt.ml a/vendor/dbuenzli-logs/src/logs_fmt.ml
index 3633b16..330aebb 100644
--- b/vendor/dbuenzli-logs/src/logs_fmt.ml
+++ a/vendor/dbuenzli-logs/src/logs_fmt.ml
@@ -28,7 +28,7 @@ let pp_header ~pp_h ppf (l, h) = match l with
 let pp_exec_header =
   let x = match Array.length Sys.argv with
   | 0 -> Filename.basename Sys.executable_name
-  | n -> Filename.basename Sys.argv.(0)
+  | _n -> Filename.basename Sys.argv.(0)
   in
   let pp_h ppf style h = Fmt.pf ppf "%s: [%a] " x Fmt.(styled style string) h in
   pp_header ~pp_h
