diff --git b/vendor/janestreet-bignum/bigint/src/dune a/vendor/janestreet-bignum/bigint/src/dune
index 856e53d..572cede 100644
--- b/vendor/janestreet-bignum/bigint/src/dune
+++ a/vendor/janestreet-bignum/bigint/src/dune
@@ -1,3 +1,3 @@
 (library (name bigint) (public_name bignum.bigint)
- (libraries core typerep num splittable_random zarith zarith_stubs_js)
- (preprocess (pps ppx_jane)))
\ No newline at end of file
+ (libraries core typerep splittable_random zarith zarith_stubs_js)
+ (preprocess (pps ppx_jane)))
